src/scheduler/globals.c

```diff
@@ -147,6 +148,19 @@ int got_sigpipe;
 
 int	second_connection;
 
+/* Stuff needed for multi-threading */
+pthread_mutex_t work_lock;
```

Comment:
@bhroam what do you think?

Reply:
You can't put global variables in a header file, only extern references to them.  We'd have to move them into a .c file.  If we think we might move this code out of the scheduler in the future, then we should treat is more like a library and move these variables elsewhere, otherwise I think they are fine where they are.  This is a file meant to hold globals.